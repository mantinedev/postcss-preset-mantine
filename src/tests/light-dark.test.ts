import { testTransform } from './utils';

const baseInput = `
.button {
  background: light-dark(var(--mantine-color-red-5), var(--mantine-color-blue-9));
}
`;

const mediaInput = `
@media screen and (min-width: 400px) {
  .button {
    background: light-dark(var(--mantine-color-green-5), var(--mantine-color-yellow-9));
  }
}
`;

const lightDarkInsideClassNameInput = `
.button-light-dark {
  color: red;
}

.button-light-dark:hover {
  color: light-dark(var(--mantine-color-red-5), var(--mantine-color-blue-9));
}
`;

const lightDarkInsideFunctionInput = `
.button {
  color: highlight-dark(red);
}
`;

const commasInput = `
.button {
  color: light-dark(rgba(0, 0, 0, 0.2), rgba(142 142 142 / 0.2));
}
`;

describe('light-dark', () => {
  it('works with base input', async () => {
    const res = await testTransform(baseInput);
    expect(res.css).toMatchSnapshot();
  });

  it('works inside media query', async () => {
    const res = await testTransform(mediaInput);
    expect(res.css).toMatchSnapshot();
  });

  it('works when light-dark is inside the class name', async () => {
    const res = await testTransform(lightDarkInsideClassNameInput);
    expect(res.css).toMatchSnapshot();
  });

  it('does not process the function when name does not match exactly', async () => {
    const res = await testTransform(lightDarkInsideFunctionInput);
    expect(res.css).toMatchSnapshot();
  });

  it('works when there are commas in the value', async () => {
    const res = await testTransform(commasInput);
    expect(res.css).toMatchSnapshot();
  });
});
