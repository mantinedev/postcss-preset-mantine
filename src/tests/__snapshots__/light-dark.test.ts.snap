// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`light-dark does not process the function when name does not match exactly 1`] = `
"
.button {
  color: highlight-dark(red);
}
"
`;

exports[`light-dark works inside media query 1`] = `
"
@media screen and (min-width: 400px) {
  .button {
    background: var(--mantine-color-green-5);
  }
    [data-mantine-color-scheme='dark'] .button {
    background: var(--mantine-color-yellow-9);
  }
}
"
`;

exports[`light-dark works when light-dark is inside the class name 1`] = `
"
.button-light-dark {
  color: red;
}

.button-light-dark:hover {
  color: var(--mantine-color-red-5);
}

[data-mantine-color-scheme='dark'] .button-light-dark:hover {
  color: var(--mantine-color-blue-9);
}
"
`;

exports[`light-dark works when there are commas in the value 1`] = `
"
.button {
  color: rgba(0, 0, 0, 0.2);
}
  [data-mantine-color-scheme='dark'] .button {
  color: rgba(142 142 142 / 0.2);
}
"
`;

exports[`light-dark works with base input 1`] = `
"
.button {
  background: var(--mantine-color-red-5);
}
  [data-mantine-color-scheme='dark'] .button {
  background: var(--mantine-color-blue-9);
}
"
`;
